Code Climate service tests failing

:clock11: **When did the problem start?**
Not sure but a while ago; I quit scrolling through the logs after one week's worth of consistent failures
<!-- Indicate when the problem started -->

:camera: **Live badge**

<!-- Provide a link to the live badge in plain text and markdown.  -->
https://img.shields.io/codeclimate/issues/angular/angular
![](https://img.shields.io/codeclimate/issues/angular/angular)

:wrench: **Is the live badge working?**

Yeah
<!-- Indicate whether or not the live badge is working. -->

:link: **CircleCI link**

https://app.circleci.com/pipelines/github/badges/daily-tests/1615/workflows/c8bf4aaf-2074-4c0e-bdc1-14ba8e5e2c7e/jobs/2517
<!-- Provide a link to the failing test in CircleCI. -->

:lady_beetle: **Stack trace**

```
CodeclimateAnalysis [mocked] issues when outer user repos query returns multiple items [ GET /issues/angular/angular.json ]
[ GET /issues/angular/angular.json ]
/home/circleci/project/shields/core/service-test-runner/cli.js

AssertionError: label mismatch: expected 'analysis' to equal 'issues'
    at Function._expectField (file:///home/circleci/project/shields/core/service-test-runner/icedfrisby-shields.js:85:51)
    at IcedFrisbyNock.<anonymous> (file:///home/circleci/project/shields/core/service-test-runner/icedfrisby-shields.js:69:26)
    at IcedFrisbyNock.<anonymous> (node_modules/icedfrisby/lib/icedfrisby.js:954:10)
    at Immediate.invokeNextHook (node_modules/icedfrisby/lib/icedfrisby.js:1003:24)
    at processImmediate (internal/timers.js:464:21)

---------------------------------------------------------------------------------------------------------------------------------------------------------


CodeclimateCoverage [mocked] test coverage when outer user repos query returns multiple items [ GET /coverage/codeclimate/codeclimate.json ]
[ GET /coverage/codeclimate/codeclimate.json ]
/home/circleci/project/shields/core/service-test-runner/cli.js

ValidationError: message mismatch: "value" with value "not found" fails to match the required pattern: /^[1-9][0-9]?%|^100%|^0%$/
    at Object.exports.process (node_modules/joi/lib/errors.js:193:16)
    at Object.internals.entry (node_modules/joi/lib/validator.js:153:26)
    at Object.exports.entry (node_modules/joi/lib/validator.js:27:30)
    at internals.Base.validate (node_modules/joi/lib/base.js:548:26)
    at Object.internals.assert (node_modules/joi/lib/index.js:225:27)
    at Object.attempt (node_modules/joi/lib/index.js:107:26)
    at Function._expectField (file:///home/circleci/project/shields/core/service-test-runner/icedfrisby-shields.js:89:13)
    at IcedFrisbyNock.<anonymous> (file:///home/circleci/project/shields/core/service-test-runner/icedfrisby-shields.js:70:26)
    at IcedFrisbyNock.<anonymous> (node_modules/icedfrisby/lib/icedfrisby.js:954:10)
    at Immediate.invokeNextHook (node_modules/icedfrisby/lib/icedfrisby.js:1003:24)
    at processImmediate (internal/timers.js:464:21)


```

:bulb: **Possible solution**

Don't know. Both of the failing tests are mocked, though worth noting that CodeClimate is a bit of an odd bird compared to our typical service request flow in that we have to make two sequential API calls in order to get data, and I believe in both of these cases that one of the two requests aren't mocked/are passing through.

Some likely useful references: #7716 and #7695

<!--- Optional: only if you have suggestions on a fix/reason for the bug -->

<!-- Love Shields? Please consider donating $10 to sustain our activities:
👉  https://opencollective.com/shields -->

--- a/services/codeclimate/codeclimate-analysis.tester.js
+++ b/services/codeclimate/codeclimate-analysis.tester.js
@@ -35,12 +35,12 @@ t.create('maintainability letter')
 t.create('issues when outer user repos query returns multiple items')
   .get('/issues/tensorflow/models.json')
   .intercept(nock =>
-    nock('https://api.codeclimate.com', { allowUnmocked: true })
+    nock('https://api.codeclimate.com')
       .get('/v1/repos?github_slug=tensorflow%2Fmodels')
       .reply(200, {
         data: [
           {
-            id: 'xxxxxxxxxxxx', // Fake repo id, which is expected to be ignored in favour of the one that does contain snapshot data.
+            id: 'xxxxxxxxxxxx', // Expected to be ignored in favour of the one that does contain snapshot data.
             relationships: {
               latest_default_branch_snapshot: {
                 data: null,
@@ -51,7 +51,7 @@ t.create('issues when outer user repos query returns multiple items')
             },
           },
           {
-            id: '57e2efacc718d40058000c9b', // Real repo id for tensorflow/models. The test retrieves live data using the real snapshot id below.
+            id: '57e2efacc718d40058000c9b',
             relationships: {
               latest_default_branch_snapshot: {
                 data: {
@@ -65,12 +65,31 @@ t.create('issues when outer user repos query returns multiple items')
             },
           },
         ],
+      })
+      .get(
+        '/v1/repos/57e2efacc718d40058000c9b/snapshots/65ae115f34117d0001055101',
+      )
+      .reply(200, {
+        data: {
+          attributes: {
+            ratings: [
+              {
+                letter: 'D',
+                measure: {
+                  value: 24.73668395092537,
+                },
+              },
+            ],
+          },
+          meta: {
+            issues_count: 11538,
+          },
+        },
       }),
   )
-  .networkOn() // Combined with allowUnmocked: true, this allows the inner snapshots query to go through.
   .expectBadge({
     label: 'issues',
-    message: Joi.number().integer().positive(),
+    message: '11538',
   })
 
 t.create('maintainability letter for non-existent repo')

--- a/services/codeclimate/codeclimate-coverage.tester.js
+++ b/services/codeclimate/codeclimate-coverage.tester.js
@@ -23,12 +23,12 @@ t.create('test coverage letter')
 t.create('test coverage when outer user repos query returns multiple items')
   .get('/coverage/codeclimate/codeclimate.json')
   .intercept(nock =>
-    nock('https://api.codeclimate.com', { allowUnmocked: true })
+    nock('https://api.codeclimate.com')
       .get('/v1/repos?github_slug=codeclimate%2Fcodeclimate')
       .reply(200, {
         data: [
           {
-            id: 'xxxxxxxxxxxx', // Fake repo id, which is expected to be ignored in favour of the one that does contain snapshot data.
+            id: 'xxxxxxxxxxxx', // Expected to be ignored in favour of the one that does contain snapshot data.
             relationships: {
               latest_default_branch_snapshot: {
                 data: null,
@@ -39,7 +39,7 @@ t.create('test coverage when outer user repos query returns multiple items')
             },
           },
           {
-            id: '558479d6e30ba034120008a9', // Real repo id for codeclimate/codeclimate. The test retrieves live data using the real test report id below.
+            id: '558479d6e30ba034120008a9',
             relationships: {
               latest_default_branch_snapshot: {
                 data: null,
@@ -53,12 +53,24 @@ t.create('test coverage when outer user repos query returns multiple items')
             },
           },
         ],
+      })
+      .get(
+        '/v1/repos/558479d6e30ba034120008a9/test_reports/65a1662cb0077b00013cb4de',
+      )
+      .reply(200, {
+        data: {
+          attributes: {
+            covered_percent: 24,
+            rating: {
+              letter: 'B',
+            },
+          },
+        },
       }),
   )
-  .networkOn() // Combined with allowUnmocked: true, this allows the inner test reports query to go through.
   .expectBadge({
     label: 'coverage',
-    message: isIntegerPercentage,
+    message: '24%',
   })
 
 t.create('test coverage percentage for non-existent repo')

